### PR TITLE
fix double scrollbar bug in WC Help dialog

### PR DESCRIPTION
**Pull request by babaissarkar**
Caused by the fact that the vertical scollbar mode in the left hand treeview was set to `never`. Setting it to `auto` allows the scrollbar to grow, and prevent the window from showing it's own scrollbar. (Issue #8576)

### CODE REVIEW

**Comment by babaissarkar** [comment]
Adding a slightly unrelated but scrolling bug to this PR :
The Achievements dialog always showed the vertical scrollbar when it wasn't needed and never showed the horizontal scrollbar at all even for extra long texts/descriptions. I've set both of these to `auto`. I've also added a fixed height to the dialog so that it doesn't hog all vertical space even in big monitors.

**Comment by Wedge009** [comment]
Thanks for this.

**Comment by stevecotton** [comment]
For UI changes, @Vultraz should be the reviewer. However, I suggest splitting this into two PRs because the achievements part needs work.

The WC part seems good. Adding a changelog entry would be nice, but otherwise I think that part is ready to merge.

The maximum height for the achievements dialog seems wrong to me - on a 1920x1080 screen (with window-manager bar at the top), cramming in into a small box in the center of the screen that only shows 5 items is worse than having a tall column that shows 10.

Making the Wesnoth window as small as it will go, a vertical double-scrollbar appears for achievements. Not a regression, this also occurs without this PR (tested on 1.18.0+dev at 439c43a59ec3d7af8d1d444394bf43c789002e73).

**Comment by babaissarkar** [comment]
@stevecotton Removed achievements changes and added changelog.

File: changelog_entries/wc-double-scrollbar-bugfix.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+### Multiplayer
```
Comment (stevecotton) [inline changelog_entries/wc-double-scrollbar-bugfix.md:1]: WC is normally under `### Campaigns` `    * World Conquest` - copy an entry from changelog.md as a starting point.

**Comment by Vultraz** [comment]
You should leave the horizontal scrollbar as "never", otherwise it could show up at times.

**Comment by babaissarkar** [comment]
> You should leave the horizontal scrollbar as "never", otherwise it could show up at times.

@Vultraz  Changed it to never.

**Review by Vultraz — APPROVED** [review]

**Comment by Wedge009** [comment]
Can we back-port this to 1.18? It's a cosmetic change that I think would be good to include in the current stable release.

File: changelog_entries/wc-double-scrollbar-bugfix.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ ### Campaigns
+    * World Conquest
+      Fix World Conquest's Help dialog showing double scrollbars due to the left tree having it's vertical scrollbar disabled. Changed both vertical and horizontal scrollbar modes to automatic. (issue #8576)
```
Comment (stevecotton) [inline changelog_entries/wc-double-scrollbar-bugfix.md:3]: It's a bit late to notice, but this changelog is out of date, it no longer changes the horizontal mode.

**Pull request merged**